### PR TITLE
DOC: stats: add realistic examples to correlation tests

### PR DESCRIPTION
**Pull request by mdhaber**
#### What does this implement/fix?
Adds a realistic example to `scipy.stats.spearmanr`. 

#### Additional information
When this looks good, I'll add similar examples to the other correlation tests.

### CODE REVIEW

**Comment by mdhaber** [comment]
This linting failure does not occur locally, and the refguide failures I got in past examples did not occur locally, either. Very weird. Anyway, I'll fix these in the next commit.

File: scipy/stats/_stats_py.py [hunk 1]
```diff
@@ -5202,72 +5202,163 @@ def spearmanr(a, b=None, axis=0, nan_policy='propagate',
...
+      and false negatives (failure to reject a false null hypothesis).
+    - Small p-values are not evidence for a *large* effect; rather, they can
+      only provide evidence for a "significant" effect, meaning that they are
+      unlikely to have occured under the null hypothesis.
+
+    Suppose that before performing the experiment, the authors had reason
+    to predict a positive correlation between the total collagen and free
+    proline measurements. In this case, they may have chosen to assess the
+    plausibility of the null hypothesis against a one-sided alternative:
+    free proline has a positive linear correlation with total collagen. In this
```
Comment (mdhaber) [inline scipy/stats/_stats_py.py:5313]: Repeated "in this case"

**Comment by j-bowhay** [comment]
> This linting failure does not occur locally

Are you using `dev.py lint` or `tools/lint_diff.py` out of interest?

File: scipy/stats/_stats_py.py [hunk 2]
```diff
@@ -5202,72 +5202,163 @@ def spearmanr(a, b=None, axis=0, nan_policy='propagate',
...
+    If the p-value is "small" - that is, if there is a low probability of
+    sampling data from independent distributions that produces such an extreme
+    value of the statistic - this may be taken as evidence against the null
+    hypothesis in favor of the alternative: the distribution of total collagen
+    and free proline are *not* independent. Note that:
+
+    - The inverse is not true; that is, the test is not used to provide
+      evidence for the null hypothesis.
+    - The threshold for values that will be considered "small" is a choice that
+      should be made before the data is analyzed [3]_ with consideration of the
```
Comment (mdhaber) [inline scipy/stats/_stats_py.py:5302]: Not reference 3 here. Check this.

File: scipy/stats/_stats_py.py [hunk 3]
```diff
@@ -5202,72 +5202,163 @@ def spearmanr(a, b=None, axis=0, nan_policy='propagate',
...
+      risks of both false positives (incorrectly rejecting the null hypothesis)
+      and false negatives (failure to reject a false null hypothesis).
+    - Small p-values are not evidence for a *large* effect; rather, they can
+      only provide evidence for a "significant" effect, meaning that they are
+      unlikely to have occured under the null hypothesis.
```
Comment (mdhaber) [inline scipy/stats/_stats_py.py:5307]:
```suggestion
      unlikely to have occurred under the null hypothesis.
```

**Comment by mdhaber** [comment]
`lint_diff.py`. and `refguide_check.py` directly.

**Comment by j-bowhay** [comment]
Ah that explains the lint failure at least. The CI job is failing because of the unicode check which is a separate script (`tools/unicode-check.py`). Running `dev.py lint` replicates what the CI job does where as `tools/lint_diff.py` is just one of the three checks that take place.

https://github.com/scipy/scipy/blob/bd120dfabb2d566b87dc1daeb4bd6ad74825141c/azure-pipelines.yml#L153-L161

**Comment by mdhaber** [comment]
Ah forgot about that. Yeah, I suppose I can try using `dev.py` for these things even if I'm still using `setup.py` to build.

File: scipy/stats/_stats_py.py (hunk 1 above)
Comment (tupui) [reply scipy/stats/_stats_py.py:5313]: I don't mind it here.

File: scipy/stats/_stats_py.py (hunk 2 above)
Comment (tupui) [reply scipy/stats/_stats_py.py:5302]: What do you mean?

**Review by tupui — APPROVED** [review]
Thanks Matt. Interesting example and reads well. I also like that you showed what does changing the alternative does.

+1 to add this to other functions.

LGTM once linter is fixed.

File: scipy/stats/_stats_py.py (hunk 2 above)
Comment (mdhaber) [reply scipy/stats/_stats_py.py:5302]: That was a note to self. I need to fix the reference number.

File: scipy/stats/_stats_py.py (hunk 1 above)
Comment (mdhaber) [reply scipy/stats/_stats_py.py:5313]: I do! : )

**Comment by mdhaber** [comment]
There were CI failures here. I skipped CI because we had discussed including other functions, and I meant to fix the CI issues when adding them.

**Comment by tupui** [comment]
My bad, some misinterpretation on my side. I fixed the issues and now CI is green.

**Pull request merged**